src/main/java/htsjdk/samtools/BAMRecordCodec.java

```diff
@@ -174,6 +181,25 @@ public void encode(final SAMRecord alignment) {
         }
     }
 
+    private void warnIfReferenceIsTooLargeForBinField(final SAMRecord rec) {
+        final SAMSequenceRecord sequence = rec.getHeader() != null ? rec.getHeader().getSequence(rec.getReferenceName()) : null;
+        if (!isReferenceSizeWarningShowed
+                && sequence != null
+                && isReferenceTooLargeForBAI(sequence)
+                && rec.getValidationStringency() != ValidationStringency.SILENT) {
+            LOG.warn("Reference length is too large for BAM bin field. Values in the bin field could be incorrect.");
+            isReferenceSizeWarningShowed = true;
+        }
+    }
+
+    private boolean isReferenceTooLargeForBAI(final SAMSequenceRecord sequence) {
+        return sequence.getSequenceLength() > GenomicIndexUtil.BIN_GENOMIC_SPAN;
+    }
```

Comment:
This method appears in two places and should probably be consolidated. See comment below.

Reply:
Done.